dandiapi/api/models/asset.py

```diff
@@ -125,25 +125,51 @@ def copy(cls, asset):
         return Asset(path=asset.path, blob=asset.blob, metadata=asset.metadata)
 
     @classmethod
-    def get_path(cls, path_prefix: str, qs: List[str]) -> Set:
+    def get_path(cls, path_prefix: str, qs: List[dict], extended: bool = False) -> List:
         """
         Return the unique files/directories that directly reside under the specified path.
 
-        The specified path must be a folder (must end with a slash).
+        Parameters
+        ----------
+        path_prefix: str
+          The specified path must be a folder (must end with a slash).
+        extended: bool, optional
+          If True, will return a list of records: original asset records or summary
+          records for directories
         """
         if not path_prefix:
             path_prefix = '/'
         prefix_parts = [part for part in path_prefix.split('/') if part]
-        paths = set()
+        paths = dict() if extended else set()
         for asset in qs:
             path_parts = [part for part in asset['path'].split('/') if part]
 
             # Pivot index is -1 (include all path parts) if prefix is '/'
             pivot_index = path_parts.index(prefix_parts[-1]) if len(prefix_parts) else -1
             base_path, *remainder = path_parts[pivot_index + 1 :]
-            paths.add(f'{base_path}/' if len(remainder) else base_path)
-
-        return sorted(paths)
+            if not extended:
+                paths.add(f'{base_path}/' if len(remainder) else base_path)
+            elif len(remainder):
+                if base_path not in paths:
+                    # Initiate the record from the asset.  It will only miss
+                    # asset_id field and otherwise be the same with cumulative information
+                    paths[base_path] = {
+                        'path': f'{base_path}/',
+                        'size': asset['size'],
+                        'created': asset['created'],
+                        'modified': asset['modified'],
```

Comment:
wouldn't it be useful to add assetid and download link here as well? mostly because someday we will show metadata of asset. adding download makes the display super fast.

Reply:
on one hand: web ui can produce those based on knowing the dataset id (just as fast). 
on the other: I do not think we even know the dataset_id or version here (in models) and download urls are dataset/version specific (to provide content-disposition).  But it could be done in the `view` portion: it could enhance those records with urls in `extended` mode indeed making web-ui really just a renderer of information and making it possible for other clients (e.g. `dandi-cli`) to readily provide that information or for API to become easier to use for similar use cases/views.  So - let's do it.
I guess for consistency with the model that field should be `contentUrl` and contain a list (of a single ATM) URL (or should we just keep it a singular string).  And it will be populated only for assets, not for directories (ATM at least, although could be some dandi:// url consumable by dandi-cli, ;-) )

Reply:
and just spotted:

> wouldn't it be useful to add assetid ...

this is for a folder -- there is no assetid for it.  For assets - records will contain it and we are just adding/returning them as is (nice and cheap ;))

Reply:
folder's don't have creaded/modified. the above code specifically is about info for an asset. so i'm assuming you have asset details here. remember, you can't construct the download path without knowing assetid. so that would at least be essential for the ui to create the path.

Reply:
> folder's don't have creaded/modified

correct. That is why I am taking min of created and max of modified across assets.  IMHO it is useful information to display. We should just document somewhere/how that it is not datetime of the folder but a "summary" across assets.

> the above code specifically is about info for an asset.

not entirely -- it traverses assets to figure out paths, including "folders/"  which are to be displayed.  It would be super useful to provide `size` for each folder, and it would come virtually "for free".

> remember, you can't construct the download path without knowing assetid. so that would at least be essential for the ui to create the path.

asset_ids are known for assets here.  code above (in view) also knows about which dandiset/version and can quickly construct contentUrl for each asset (and skip "folder/" paths records of which will lack "asset_id")

Reply:
> That is why I am taking min of created and max of modified across assets.

i think that's happening in a different branch of this else. where i put this comment, i don't think asset uuid is being added. this code comment is specific to a single asset not a folder.

Reply:
it is impossible to comprehend now the indentation levels due to this comment, here is the code snippet:
```python
            if not extended:
                paths.add(f'{base_path}/' if len(remainder) else base_path)
            elif len(remainder):
                if base_path not in paths:
                    # Initiate the record from the asset.  It will only miss
                    # asset_id field and otherwise be the same with cumulative information
                    paths[base_path] = {
                        'path': f'{base_path}/',
                        'size': asset['size'],
                        'created': asset['created'],
                        'modified': asset['modified'],
                    }
                else:
                    # update record
                    rec = paths[base_path]
                    # Assuming we store in UTC but might better
                    rec['size'] += asset['size']
                    rec['created'] = min(rec['created'], asset['created'])  # earliest
                    rec['modified'] = max(rec['modified'], asset['modified'])  # latest
            else:
                # asset
                paths[base_path] = asset
```
I think it is correct.  First block -- no extended, 2nd - extended directory, 3rd - extended asset

Reply:
ok  yes. i see it better now.